### PR TITLE
Skillup books

### PR DESCRIPTION
**Pull request by Omnione**
Added all skillup books and made them functional.

This has been tested thoroughly on Canaria when it was on Topaz.

Now its been redone for Topaz-Next refactor release.

**_I affirm:_**
- [X] that I agree to Topaz Next's [Limited Contributor License Agreement](https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [X] that I have read the [Contributing Guide](https://github.com/topaz-next/topaz/blob/release/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/topaz-next/topaz/blob/release/CODE_OF_CONDUCT.md)
- [X] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: scripts/globals/item_utils.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return
+    end
+
+    target:messageBasic(38, skill, skillGain)
+    
+    if math.floor(skillLevel/10) < math.floor((skillLevel + skillGain)/10) then
+        target:messageBasic(53, skill, skillLevel/10)
+    end
+
+    return target:setSkillLevel(skill, ((skillLevel + skillGain)/10)*10)
```
Comment (TeoTwawki) [inline scripts/globals/item_utils.lua]: `trySkillUp(skillType, targetLvl)` can save you from all that diff checking, pretty sure.

I summon thee @claywar

**Comment by Omnione** [comment]
gonna leave this as a draft while we wait on Clays Pr, then ill get it changed to the trySkillUp()

File: scripts/globals/item_utils.lua (hunk 1 above)
Comment (ghost) [reply scripts/globals/item_utils.lua]: Are you sure?!  Summoned!  That binding was added during the sol refactor, just put up #2660 to support a forced skillup opportunity on call.

Comment (Omnione) [reply scripts/globals/item_utils.lua]: using trySkillUp now

**Comment by Omnione** [comment]
oh, need a new line on item_utils

File: scripts/globals/item_utils.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    local mainCap = target:getMaxSkillLevel(target:getMainLvl(), target:getMainJob(), skill) or 0
+    local subCap = target:getMaxSkillLevel(target:getSubLvl(), target:getSubJob(), skill) or 0
+    local skillLevel = target:getCharSkillLevel(skill)/10
+    local mainDif = (mainCap*10)/10 - (target:getCharSkillLevel(skill)*10)/100
+    local subDif = (mainCap*10)/10 - (target:getCharSkillLevel(skill)*10)/100
+    local noSkill = 0
+    local canUse = 0
+
+    if mainCap == 0 then
+        noSkill = noSkill +1
```
Comment (ghost) [inline scripts/globals/item_utils.lua]: Can you please add a space after +?  (repeats below)

**Review by ghost — COMMENTED** [review]
Content looks good to me, there's a couple of spacing issues (and maybe the mainDif/subDif calculations?), but overall nice work. 👍

**Comment by ghost** [comment]
Note to future self: Apparently book skillups can also affect sub-job skills.  Need to adjust `charutils::TrySkillUp` and the binding to be able to support this.

File: scripts/globals/item_utils.lua (hunk 2 above)
Comment (Omnione) [reply scripts/globals/item_utils.lua]: all done done

**Review by ghost — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**